src/test/java/org/apache/maven/buildcache/its/junit/IntegrationTestExtension.java

```diff
@@ -208,6 +208,8 @@ public Object resolveParameter( ParameterContext parameterContext,
 
             Verifier verifier = new Verifier( testExecutionDir.toString(), true );
             verifier.setLogFileName( "../log.txt" );
+            verifier.setSystemProperty( "projectVersion", System.getProperty( "projectVersion" ) );
+            verifier.setLocalRepo( System.getProperty( "localRepo" ) );
```

Comment:
Please notice I also faced a similar problem [while adding tests](https://github.com/apache/maven-build-cache-extension/pull/36/src/test/java/org/apache/maven/buildcache/its/ForkedExecutionsTest.java). But I decided to use per-run temp directory to have clean state between runs (at least for cache).

So here are the questions:

- how this repository is cleaned?
- Will it be unique between tests? Each integration test assumes clean state
- On local workstation - will it be also clean each time? Tests to be run on local machine as well

Reply:
- `localRepo` is used to set property `maven.repo.local` which point to local Maven repository, instead of standard `.m2/repository`

- `localRepo` value  is passed from pom.xml

```
<localRepo>${project.build.directory}/local-repo</localRepo>
```

so it is in `target` directory, it will be cleaned in standard way

`localRepo` is first populated by m-invoker-p by install goal before integration test will be started, and current project arifact is instaled, next Maven process which is executed during test, download everything else was is needed

before my change is m-install-p was configured to install project artifact in standard Maven local repository

it is shared by all test

on local workstation it will be the same

https://maven.apache.org/plugins/maven-invoker-plugin/install-mojo.html

Reply:
Thanks for clarifications!